code/scripting/api/objs/hudconfig.cpp

```diff
@@ -0,0 +1,237 @@
+#include "hudconfig.h"
+
+#include "hud/hudconfig.h"
+#include "ui/ui.h"
+#include "scripting/api/objs/color.h"
+#include "iff_defs/iff_defs.h"
+#include "globalincs/alphacolors.h"
+
+namespace scripting {
+namespace api {
+
+gauge_config_h::gauge_config_h() : gauge(-1) {}
+gauge_config_h::gauge_config_h(int l_gauge) : gauge(l_gauge) {}
+
+HC_gauge_region* gauge_config_h::getGauge() const
+{
+	return &HC_gauge_regions[GR_1024][gauge];
```

Comment:
`getGauge` and `getName` should check that `gauge` is in range

Reply:
This is redundant, but whatever.

Reply:
The `isValid` function is sufficient